File: src/main/java/gr/apt/security/TokenGenerator.java
package gr.apt.security;

import io.smallrye.jwt.build.Jwt;

import javax.enterprise.context.ApplicationScoped;
import java.util.Arrays;
import java.util.HashSet;

@ApplicationScoped
public class TokenGenerator {

    public String generateToken(String username) {
        return Jwt.issuer("https://example.com/issuer")
                //.upn("jdoe@quarkus.io")
                .groups(new HashSet<>(Arrays.asList("User", "Admin")))
                .claim("username", username)
                .sign();
    }
}

File: src/main/java/gr/apt/lms/config/security/TokenGenerator.kt
package gr.apt.lms.config.security

import io.smallrye.jwt.build.Jwt
import java.util.*
import javax.enterprise.context.ApplicationScoped

@ApplicationScoped
object TokenGenerator {

    internal const val USERNAME_CLAIM = "username"
    fun generateToken(username: String): String {
        //TODO: make generate token work for real time service
        return Jwt.issuer("https://example.com/issuer") //.upn("jdoe@quarkus.io")
            .groups(HashSet(Arrays.asList("User", "Admin")))
            .claim(USERNAME_CLAIM, username)
            .sign()
    }
}